main/src/library/Filterable.flix

```diff
@@ -17,16 +17,17 @@
 ///
 /// A trait for filtering container functors.
 ///
-pub trait Filterable[m : Type -> Type] with Functor[m] {
+pub trait Filterable[m : Type -> Type] with Functor[m], Foldable[m] {
 
     ///
     /// Applies the partial function `f` to every element in `x` collecting the results.
     ///
-    pub def filterMap(f: a -> Option[b] \ ef, t: m[a]): m[b] \ ef
+    pub def filterMap(f: a -> Option[b] \ ef, t: m[a]): m[b] \ (ef + aef) where Foldable.Aef[m] ~ aef
```

Comment:
Is there a problem with `(ef + Foldable.Aef[m])`?

Reply:
Same error for both ways - I tried with `(ef + Foldable.Aef[m])` first then wondered if making a variable would alleviate it.

Reply:
I see.

This PR will hopefully fix the issue for you: https://github.com/flix/flix/pull/7686

Reply:
Thanks Matthew

Reply:
PR with fix merged.

Reply:
Answering the top question I've reverted back to `(ef + Foldable.Aef[m])`